[search-documents] Migrated Samples to the new pipeline

sarangan12 opened the pull request:
The samples migration task for search-documents has not been migrated yet. This has caused CI failures on a daily basis - https://dev.azure.com/azure-sdk/internal/_build?definitionId=1650&_a=summary. This PR migrates all the samples and confirms that the CI is working fine. I have created a manual run for the Tests CI - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1035904&view=results. 

@witemple-msft Please review and approve the PR.

witemple-msft commented on sdk/search/search-documents/tsconfig.json:
```diff
@@ -2,7 +2,11 @@
+    }
   },
-  "exclude": ["node_modules", "types", "temp", "browser", "dist", "dist-esm", "./samples/**/*.ts"]
+  "exclude": ["node_modules", "types", "temp", "browser", "dist", "dist-esm"],
```
You can remove `"excludes"`, since none of these overlap with the `includes` it has no effect.

sarangan12 replied:
Done

witemple-msft commented on sdk/search/search-documents/samples-dev/bufferedSenderAutoFlushSize.ts:
```diff
@@ -5,22 +5,26 @@ import {
 import * as dotenv from "dotenv";
 dotenv.config();
 
+/**
```
Such a small thing, but I think these are nicer if they are at/near the top of the file.

sarangan12 replied:
Done

witemple-msft reviewed (commented):
Couple things:

1. We need to regenerate the published copies of the samples with `npx dev-tool samples generate`. I see deleted sample files but no new generated copies.
2. We're missing copyright headers on the samples, we need to have:

```
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
```

at the top of all the sample files.

sarangan12 commented:
@witemple-msft Thanks for the feedback. I have incorporated all the comments. 

> 1. We need to regenerate the published copies of the samples with `npx dev-tool samples generate`. I see deleted sample files but no new generated copies.

I am assuming you meant `npx dev-tool samples publish`. I have executed publish command and updated the results in the latest commit. Please take a look. Thanks

witemple-msft reviewed (approved):
All LGTM. Thank you!
